dockerRunner.go

```diff
@@ -1087,3 +858,232 @@ func (dr *dockerRunnerImpl) DeleteBridgeNetwork(ctx context.Context) error {
 
 	return nil
 }
+
+func (dr *dockerRunnerImpl) generateEntrypointScript(shell string, commands []string, commandsNotAsJob bool) (path string, extension string, err error) {
+
+	r, _ := regexp.Compile("[a-zA-Z0-9_]+=|;|\\||&&|\\|\\|")
+
+	firstCommands := []struct {
+		Command         string
+		RunInBackground bool
+	}{}
+	for _, c := range commands[:len(commands)-1] {
+		// check if the command is assigning a value, in which case it shouldn't be run in the background
+		match := r.MatchString(c)
+		runInBackground := !commandsNotAsJob && !match
```

Comment:
can we call the variable `commandsAsJob`? The double negative here is confusing me.

Reply:
I've already renamed it to `runCommandsInForeground`